cmd/bucket-replication.go

```diff
@@ -1240,6 +1240,11 @@ func (ri ReplicateObjectInfo) replicateObject(ctx context.Context, objectAPI Obj
 		return
 	}
 
+	// Set the encrypted size for SSE-C objects
+	if crypto.SSEC.IsEncrypted(objInfo.UserDefined) {
+		size = objInfo.Size
```

Comment:
for SSE-C encrypted objects which are compressed the correct method is GetActualSize() are you sure this fix is correct?

Reply:
After site replication enabling and loading the object, even if object was getting replicated fine with ssec, it was showing and error
```
API: SYSTEM.replication
Time: 10:33:32 UTC 06/06/2024
DeploymentID: 7a1b9b43-562e-47d6-9784-0a23c5fcc118
Error: unable to replicate for object test-bucket/hello.txt(2e68eb65-a600-4dcd-bb7d-b303462bcb7e): to (target: https://localhost:9010): Put "https://localhost:9010/test-bucket/hello.txt?versionId=2e68eb65-a600-4dcd-bb7d-b303462bcb7e": net/http: HTTP/1.x transport connection broken: http: ContentLength=16 with Body length 48 (*fmt.wrapError)
       4: internal/logger/logger.go:268:logger.LogIf()
       3: cmd/logging.go:16:cmd.replLogIf()
       2: cmd/bucket-replication.go:1308:cmd.ReplicateObjectInfo.replicateObject()
       1: cmd/bucket-replication.go:1084:cmd.replicateObject.func2()
```

Reply:
the reason this would be a bit wrong as it doesn't send the `size` original size of the object for the event notification - you are overriding that with SSE-Cs encrypted size.

Reply:
While manual testing I faced this issue. May be test covering one path.

Reply:
no it must be done before `GetActualSize()` since you are not intested in the acutla size of the object.

Reply:
you mean before L#1230, right?